src/tapis-hooks/files/useUpload.ts

```diff
@@ -0,0 +1,78 @@
+import { useState, useCallback } from 'react';
+import { useMutation, MutateOptions } from 'react-query';
+import { Files } from '@tapis/tapis-typescript';
+import { insertAxios as insert } from 'tapis-api/files';
+import { useTapisConfig } from 'tapis-hooks';
+import QueryKeys from './queryKeys';
+
+export type InsertHookParams = {
+  systemId: string;
+  path: string;
+  file: File;
+};
+
+const useUpload = () => {
+  const { basePath, accessToken } = useTapisConfig();
+  const jwt = accessToken?.access_token || '';
+
+  const [progress, setProgress] = useState<number>(0);
+  const [uploadingFile, setUploadingFile] = useState<File | undefined>(
+    undefined
+  );
+
+  const getProgress = useCallback(() => {
```

Comment:
I think the progress needs to go to the Uploader client. The reason is that we may want to stream multiple uploads concurrently, and the hook can't track each upload's progress state independently if they run in parallel. The good news is that the reducer state can be expanded to include progress and the callback that is passed in for the upload function can just be

```
onProgress: (progress) => dispatch({key: file.path!, event: 'progress', progress })
```

or something. It can be more redux-like.